src/main/java/io/vertx/core/net/impl/NetClientImpl.java

```diff
@@ -280,7 +280,7 @@ protected void finalize() throws Throwable {
     // Make sure this gets cleaned up if there are no more references to it
     // so as not to leave connections and resources dangling until the system is shutdown
     // which could make the JVM run out of file handles.
-    close((Handler<AsyncResult<Void>>) Promise.<Void>promise());
+    closeFuture.close();
```

Comment:
I'd remove the method?

Reply:
That would be my preference, but it is up to @vietj . If he agrees I will remove both of them.

Reply:
@Sanne we should at some point get rid of finalizer but we need to keep this cleanup. I know that Java 9 has a cleaner that can be used by Vert.x still relies on Java 8.

Reply:
I understand and no rush :)

But I suppose you know well that a finalize() method is not guaranteed to be invoked, and could be invoked multiple times? In case you prefer to keep it, the methods it calls should at least be idempotent, so to allow a double-close event.

I haven't tried it but I think you should be able to replace any finalizer with a ReferenceQueue  + PhantomReference combination, which would give you:

1. better quarantees of it being invoked at all
2. still compatible with Java 8
3. will also work when finalize() no longer works

Might be worth a shot? No pressure, just writing down the idea for future.